examples/03_connectivity/plot_compare_decomposition.py

```diff
@@ -34,64 +40,103 @@
 print('First functional nifti image (4D) is at: %s' %
       rest_dataset.func[0])  # 4D data
 
-###############################################################################
-# Create two decomposition estimators
-# ------------------------------------
-from nilearn.decomposition import DictLearning, CanICA
 
-n_components = 40
-
-###############################################################################
-# Dictionary learning
-# --------------------
-#
+####################################################################
+# Here we apply CanICA on the data
+# ---------------------------------
 # We use as "template" as a strategy to compute the mask, as this leads
 # to slightly faster and more reproducible results. However, the images
 # need to be in MNI template space
-dict_learning = DictLearning(n_components=n_components,
+
+from nilearn.decomposition import CanICA
+
+canica = CanICA(n_components=20,
+                memory="nilearn_cache", memory_level=2,
+                threshold=3.,
+                n_init=1,
+                verbose=10,
+                mask_strategy='template',
+                random_state=0)
+canica.fit(func_filenames)
+
+# Retrieve the independent components in brain space. Directly
+# accesible through attribute `components_img_`.
+canica_components_img = canica.components_img_
+# components_img is a Nifti Image object, and can be saved to a file with
+# the following line:
+canica_components_img.to_filename('canica_resting_state.nii.gz')
+
+
+####################################################################
+# To visualize we plot the outline of all components on one figure
+# -----------------------------------------------------------------
+from nilearn.plotting import plot_prob_atlas
+
+# Plot all ICA components together
+plot_prob_atlas(canica_components_img, title='All ICA components')
+
+
+####################################################################
+# Finally, we plot the map for each ICA component separately
+# -----------------------------------------------------------
+from nilearn.image import iter_img
+from nilearn.plotting import plot_stat_map, show
+
+for i, cur_img in enumerate(iter_img(canica_components_img)):
+    plot_stat_map(cur_img, display_mode="z", title="IC %d" % i,
+                  cut_coords=1, colorbar=False)
+
+show()
```

Comment:
Show should be used only at the end of the script, as it can be a blocking command.